include/engine/datafacade/internal_memory_datafacade.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef INTERNAL_DATAFACADE_HPP
+#define INTERNAL_DATAFACADE_HPP
+
+// implements all data storage when shared memory is _NOT_ used
+
+#include "engine/datafacade/bigramblock_datafacade_base.hpp"
+#include "storage/storage.hpp"
+
+namespace osrm
+{
+namespace engine
+{
+namespace datafacade
+{
+
+/**
+ * This datafacade uses a process-local memory block to load
+ * data into.  The structure and layout is the same as when using
+ * shared memory, so this class, and the SharedDatafacade both
+ * share a common base.
+ * This class holds a unique_ptr to the memory block, so it
+ * is auto-freed upon destruction.
+ */
+class InternalDataFacade final : public BigRAMBlockDataFacadeBase
+{
+
+  private:
+    std::unique_ptr<char[]> internal_memory;
```

Comment:
so the only difference between the `internal` data facade and the `shared datafacade` is where `internal_memory` / `shared_memory` is stored?
From my understanding this would be a property of the storage, and the `internal` data should in that case be resident to an `internal storage` and not be part of the facade. The facade describes how to access the data, the storage how it is stored.

Previously I already stated that I don't get the separation of both. After all, the datafacade is what we need to access the storage data. So why are we decoupling the data and its access patterns?